From 2a15d833300f911ad0f6d90896b67a894074ee18 Mon Sep 17 00:00:00 2001
From: Ganesh Maharaj Mahalingam <ganesh.mahalingam@intel.com>
Date: Thu, 27 Jan 2022 11:24:43 -0800
Subject: [PATCH] Enable journald logging

Signed-off-by: Ganesh Maharaj Mahalingam <ganesh.mahalingam@intel.com>
---
 meson.build | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 7e00979..1b2b850 100644
--- a/meson.build
+++ b/meson.build
@@ -7,6 +7,7 @@ project('conmon', 'c',
         meson_version : '>= 0.46',
 )
 
+
 git_commit = ''
 
 git = find_program('git', required : false)
@@ -32,6 +33,8 @@ add_project_arguments('-Os', '-Wall', '-Werror',
                       '-DVERSION=' + conf.get('VERSION'),
                       '-DGIT_COMMIT=' + conf.get('GIT_COMMIT'),
                       language : 'c')
+add_project_arguments('-DUSE_JOURNALD=1', language : 'c')
+incdir = include_directories('/usr/include')
 
 glib = dependency('glib-2.0')
 
@@ -43,6 +46,8 @@ else
   libdl = cc.find_library('dl')
 endif
 
+systemd = dependency('libsystemd')
+
 executable('conmon',
            ['src/conmon.c',
             'src/config.h',
@@ -78,7 +83,8 @@ executable('conmon',
             'src/utils.h',
             'src/seccomp_notify.c',
             'src/seccomp_notify.h'],
-           dependencies : [glib, libdl],
+           dependencies : [glib, libdl, systemd],
            install : true,
            install_dir : join_paths(get_option('libexecdir'), 'podman'),
+           include_directories: incdir,
 )
-- 
2.35.1

